mm2src/coins/z_coin/z_coin_errors.rs

```diff
@@ -23,6 +23,7 @@ pub enum UpdateBlocksCacheErr {
     UtxoRpcError(UtxoRpcError),
     InternalError(String),
     JsonRpcError(JsonRpcError),
+    ClientIterError(String),
```

Comment:
The reason I created  `ClientIterError` with `String` is, because we have two types of error during `scan_blocks` function:[`tonic::Status` ](https://github.com/KomodoPlatform/atomicDEX-API/commit/405d183b0b0259e04b001526d01044305f85ff62#diff-878c719c6e82f1f4290cc74719c8a7de8e3ef2d5927ebbee5925e3dd54655cfcR109-R110) and [`MmError<UpdateBlocksCacheErr>>`](https://github.com/KomodoPlatform/atomicDEX-API/commit/405d183b0b0259e04b001526d01044305f85ff62#diff-878c719c6e82f1f4290cc74719c8a7de8e3ef2d5927ebbee5925e3dd54655cfcR119-R120). So we need to keep them together in the result error message.

NOTE: after this comment I will push commit with `err.into_inner()` for `MmError<UpdateBlocksCacheErr>>`, cose later I will wrap error into MmError anyway.

So as if I have to have `ClientIterError(String)`, I decided to use it for `tonic::Status`. And now I see, that in this case I can remove `GrpcError` from `Enum`. So, the first solution is to keep code of errors as we have now.

There is another way, I can live `ClientIterError(String)` for vector of errors from `scan_blocks` function and use `GrpcError(Vec<tonic::Status>)` for `get_block_height` function. Thus, we have homogeneous errors, and we can just implement `From<Vec<tonic::Status>> for UpdateBlocksCacheErr`. However, `Vec<tonic::Status>` cannot be formatted with the default formatter. And I cannot implement `Display` only for `Vec<tonic::Status>`

Due to this, I will remove `Display` from derive(we will have conflicts) and implement `Display` manually, just like that (I checked it works)

```rust
impl fmt::Display for UpdateBlocksCacheErr {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match *self {
            UpdateBlocksCacheErr::GrpcError(ref err) => {
                for e in err {
                    write!(f, "\t{}", e)?;
                }
                Ok(())
            },
            UpdateBlocksCacheErr::BlocksDbError(ref err) => write!(f, "{}", err),
            UpdateBlocksCacheErr::ZcashSqliteError(ref err) => write!(f, "{}", err),
            UpdateBlocksCacheErr::UtxoRpcError(ref err) => write!(f, "{}", err),
            UpdateBlocksCacheErr::InternalError(ref err) => write!(f, "{}", err),
            UpdateBlocksCacheErr::JsonRpcError(ref err) => write!(f, "{}", err),
            UpdateBlocksCacheErr::ClientIterError(ref err) => write!(f, "{}", err),
        }
    }
}
```

Reply:
Upd I decided, we can keep `GrpcError(Vec<tonic::Status>)` for cases, when we have only one error type during iteration.